rubyfmt.rb

```diff
@@ -205,34 +205,74 @@ def on_regexp_literal(*args)
 
 GC.disable
 
-def parse_file(fn)
-  file_data = File.read(fn)
-  [file_data, Parser.new(file_data).parse]
+def parse_file(file_data)
+  Parser.new(file_data).parse
 end
 
-def rubyfmt_dir(dn)
-  files = Dir[File.join(dn, "**/*.rb")]
-  files.each do |glob_fn|
-    file_data, parsed = parse_file(glob_fn)
-    Rubyfmt::format_to_file(glob_fn, file_data, parsed)
+def normalize_dir(path)
+  path.end_with?('/') ? path : "#{path}/"
+end
+
+class InvalidByteSequenceError < ArgumentError
+  RE = /invalid byte sequence/.freeze
+  def self.===(error)
+    error.is_a?(ArgumentError) && error.message.match?(RE)
   end
 end
 
+RUBY_PATHS_PATTERN = "**/{*.rb,*.builder,*.jbuilder,*.ru,*.rake,*.gemspec,Gemfile,Capfile,capfile,Guardfile,.Guardfile,rakefile,Rakefile,Rake,.irbrc,irb.rc,_irbrc,$irbrc,pryrc,.pryrc}"
+SHEBANG_PATTERN = /\A#!.*\bruby\b/
+def rubyfmt_dir(path)
+  Dir.new(path).each do |child|
+    next if child == '.' || child == '..'
```

Comment:
if this project doesn't run on 2.4 or earlier then this line could be removed and the previous changed to `Dir.each_child do |child|`